local/nginx/include/routes.conf

```diff
@@ -1,29 +1,145 @@
 location ~ ^/content/(.*)$ {
+
+    set_by_lua_file $secret "/etc/nginx/lua/read_public_key.lua";
+    set_by_lua_file $missing_token "/etc/nginx/lua/missing_token.lua";
+    set_by_lua_file $invalid_token "/etc/nginx/lua/invalid_token.lua";
+    set_by_lua_file $jwt_req_zone "/etc/nginx/lua/parse_jwt_as_token.lua" "content/" $1;
+
+    proxy_pass_request_headers  on;
+    proxy_connect_timeout       600;
+    proxy_send_timeout          600;
+    proxy_read_timeout          600;
+    send_timeout                600;
+    proxy_set_header X-Forwarded-Host  $host;
+
+    # Hacky workaround as nginx does not allow redirects to named locations under if conditional 
+    error_page 400 = @uniqueLimitContentServer;
+
+    if ($missing_token) {
+        # It does not return an error code
+        return 400;
+        break;
+    }
+
+    if ($invalid_token) {
+        return 401 "Access denied because JWT is expired or invalid";
+        break;
+    }
+
+    limit_req zone=zone_with_token;
     proxy_pass http://content-server/$1$is_args$args;
+}
+
+
+location @uniqueLimitContentServer {
+    set_by_lua_block $no_key { return "noKey" }
+    limit_req zone=no_token;
+
     proxy_pass_request_headers  on;
     proxy_connect_timeout       600;
     proxy_send_timeout          600;
     proxy_read_timeout          600;
     send_timeout                600;
     proxy_set_header X-Forwarded-Host  $host;
+
+    proxy_pass http://content-server/$1$is_args$args;
 }
 
+
 location  ~ ^/comms/(.*)$ {
+    set_by_lua_file $secret "/etc/nginx/lua/read_public_key.lua";
+    set_by_lua_file $missing_token "/etc/nginx/lua/missing_token.lua";
+    set_by_lua_file $invalid_token "/etc/nginx/lua/invalid_token.lua";
+    set_by_lua_file $jwt_req_zone "/etc/nginx/lua/parse_jwt_as_token.lua" "comms/" $1;
+
+    proxy_pass_request_headers  on;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header Upgrade $http_upgrade;
+    proxy_set_header Connection "Upgrade";
+
+    # Hacky workaround as nginx does not allow redirects to named locations under if conditional 
+    error_page 400 = @uniqueLimitCommsServer;
+
+    if ($missing_token) {
+        # It does not return an error code
+        return 400;
+        break;
+    }
+
+    if ($invalid_token) {
+        return 401 "Access denied because JWT is expired or invalid";
+        break;
+    }
```

Comment:
To prevent accidental divergence of use cases and to reuse configs and duplicated code: is there a way to reuse this snippets across files?

Reply:
I'll try, anyway configuring Nginx is really tricky.